contents/java/JUnit5.md

```diff
@@ -0,0 +1,259 @@
+
+## Why do automated testing?
+There are many reasons why automated testing is important for any software engineering project.
+
+<trigger for="modal:automatedTesting2103" trigger="click">Automated testing</trigger>
+* is the first line of defence against bugs
+* protects against <popover effect="fade" content="changes that unintentionally break existing functionality" placement="top">regression</popover>
+* checks against expected output, preventing errors arising from <popover effect="fade" content="looking manually at output" placement="top">LMAO</popover> or <popover effect="fade" content="looking manually for abnormal output" placement="top">LMFAO</popover>. ([Acronym source](https://stackoverflow.com/questions/10858990/why-use-junit-for-testing))
+* ultimately, gives the developer confidence that their code works correctly.
+
+In fact, testing is so powerful and important that an entire <trigger for="modal:agileDev" trigger="click">agile development process</trigger> named <trigger for="modal:tdd" trigger="click">Test Driven Development</trigger> is designed with tests as the centerpiece of software development.
+
+## Why use JUnit 5?
+
+### Benefit: JUnit 5 is the most popular testing framework for Java
+According to [Synk's 2018 report](https://snyk.io/blog/jvm-ecosystem-report-2018-tools/), JUnit is the most popular testing framework for software engineers working in Java. Surveying over 10,200 respondents, an impressive 78% of respondents use JUnit as their primary testing framework.
+<pic src="{{baseUrl}}/contents/java/junit5_popularity.png"></pic>
+
+### Benefit: JUnit 5 is tightly integrated with IntelliJ
+JUnit is tightly integrated with IntelliJ, the most popular IDE (Source: [Synk's 2018 report](https://snyk.io/blog/jvm-ecosystem-report-2018-tools/)) for Java.
+
+When using JUnit 5 on IntelliJ, you can
+* Run tests by simply clicking a button
+* Run a subset of tests, to test specific functionality
+* Analyse <popover effect="fade" content="percentage of code checked by tests" placement="top">code coverage</popover>
+* Compare the difference between expected and actual result
+
+### Benefit: JUnit 5 is an [open source project](https://github.com/junit-team/junit5/)
+Being an open source project, JUnit 5 actively accepts feedback and contributions from the community. New features are released on cycles averaging every 3-5 weeks. The fast development cycle means that JUnit can constantly keep up with new Java features.
+For example, the <popover effect="fade" content="To test filesystems in a isolated environment" placement="top">Temporary Directory</popover> feature was introduced in JUnit 5.4.0 after receiving <a href="https://github.com/junit-team/junit5/issues/1247">feedback</a> on its usefulness.
```

Comment:
This too is not that noteworthy. In fact, most dev tools are open source tools these days.